lib/src/task_impl.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:meilisearch/src/task.dart';
+
+import 'index_impl.dart';
+import 'task_info.dart';
+
+class TaskImpl implements TaskInfo {
+  final int updateId;
+  final MeiliSearchIndexImpl index;
+
+  TaskImpl(this.index, this.updateId);
+
+  factory TaskImpl.fromMap(
+    MeiliSearchIndexImpl index,
+    Map<String, dynamic> map,
+  ) =>
+      TaskImpl(index, map['uid'] as int);
+
+  @override
+  Future<Task> getStatus() async {
```

Comment:
I ask this question somewhere else but I'm not sure to know why this `method` is called `getStatus`?

Reply:
This method is used to get the task status from the server, is currently used inside the `extension TaskWaiter` in the tests.

Reply:
Ok thanks